File: 0392-is-subsequence/0392-is-subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int c1 = 0;
        int c2 = 0;
        
        while(c1 < s.size() && c2 < t.size()){
            if(s[c1] == t[c2]){
                c1++;
            }
            c2++;
        }
        
        return c1 == s.size();
    }
};

File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        size = 0
        
        if(len(s) == 0 and len(t) == 0):
            return True
        
        if(len(t) == 0):
            return False
        
        if(len(s) == 0):
            return True
        
        for i in t:
            if(i == s[size]):
                size+=1
            
            if(size == len(s)):
                return True
            # print(i, s[size])
            # print(i)
        
        
        return size==len(s)
        